src/mjolnir/util.cc

```diff
@@ -47,6 +47,36 @@ std::string remove_double_quotes(const std::string& s) {
   return ret;
 }
 
+// Compute a curvature metric given an edge shape
+uint32_t compute_curvature(const std::list<PointLL>& shape) {
+  // Edges with just 2 shape points have no curvature.
+  // TODO - perhaps a post-process to "average" curvature along adjacent edges
+  // and smooth curvature on connected edges may be desirable?
+  if (shape.size() == 2) {
+    return 0;
+  }
+
+  // Iterate through sets of shape vertices and compute a radius of curvature.
+  // Apply a score to each section.
+  uint32_t n = 0;
+  float total_score = 0.0f;
+  auto p1 = shape.begin();
+  auto p2 = p1; p2++;
+  auto p3 = p2; p3++;
+  for (; p3 != shape.end(); ++p1, ++p2, ++p3) {
+    float radius = p1->Curvature(*p2, *p3);
+    if (!std::isnan(radius)) {
+      // Compute a score and cap it at 25 (that way one sharp turn doesn't
```

Comment:
Whether this is desirable might strongly depend on how the `curvature` property is intended to be used.  If I'm driving a truck, one sharp turn may make an otherwise gently curved road impassible.

Is there an enumeration of the problems that a general measure like this will address?

Reply:
I was thinking of this too. This point may argue for a second "minimum curve radius" parameter that would be used for strict avoidance of an edge for some vehicle parameters, perhaps in combination with lane count and road class. The curvature parameter here I think would be used as a weighting factor to favor or avoid curved roads: motorcycles may like curvy roads but RVs may not like them. Also, modulating speed based on curvature can be useful in some cases in addition to the weighting factor. For example, curved descents on a bike would be slower than straighter descents (at least for most cyclists!).